tests/Integration/Classes/ObjectModelTest.php

```diff
@@ -0,0 +1,691 @@
+                    'name' => $localizedNames,
+                ],
+            ],
+        ];
+
+        yield [
+            $initialValues,
+            [self::DEFAULT_SHOP_PLACEHOLDER, self::SECOND_SHOP_PLACEHOLDER],
+            [
+                self::DEFAULT_SHOP_PLACEHOLDER => [
+                    'name' => $updatedLocalizedNames,
+                    'enabled' => false,
+                ],
+                self::SECOND_SHOP_PLACEHOLDER => [
+                    'name' => $updatedLocalizedNames,
+                    'enabled' => false,
+                ],
+            ],
+            [
+                self::DEFAULT_SHOP_PLACEHOLDER => [
+                    'name' => [
+                        self::DEFAULT_LANGUAGE_PLACEHOLDER => true,
+                        self::SECOND_LANGUAGE_PLACEHOLDER => true,
+                    ],
+                ],
+                self::SECOND_SHOP_PLACEHOLDER => [
+                    'enabled' => true,
+                ],
+            ],
+            [
+                self::DEFAULT_SHOP_PLACEHOLDER => [
+                    'enabled' => 1,
+                    'name' => $updatedLocalizedNames,
+                ],
+                self::SECOND_SHOP_PLACEHOLDER => [
+                    'enabled' => 0,
+                    'name' => $localizedNames,
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @param TestableObjectModel $object
+     * @param array $expectedProperties
+     */
+    private function checkObjectFields(TestableObjectModel $object, array $expectedProperties): void
+    {
+        foreach ($expectedProperties as $field => $expectedValue) {
+            if (is_array($expectedValue)) {
+                $expectedValue = $this->convertLocalizedValue($expectedValue);
+            }
+            $this->assertEquals($expectedValue, $object->{$field});
+        }
+    }
+
+    /**
+     * @param TestableObjectModel $object
+     * @param array $updatedProperties
+     */
+    private function applyModifications(TestableObjectModel $object, array $updatedProperties): void
+    {
+        foreach ($updatedProperties as $field => $value) {
+            if (is_array($value)) {
+                $object->{$field} = $this->convertLocalizedValue($value);
+            } else {
+                $object->{$field} = $value;
+            }
+        }
+    }
+
+    /**
+     * @param array $value
+     *
+     * @return array
+     */
+    private function convertLocalizedValue(array $value): array
+    {
+        $localizedValue = [];
+        if (isset($value[self::DEFAULT_LANGUAGE_PLACEHOLDER])) {
+            $localizedValue[$this->defaultLanguageId] = $value[self::DEFAULT_LANGUAGE_PLACEHOLDER];
+        }
+        if (isset($value[self::SECOND_LANGUAGE_PLACEHOLDER])) {
+            $localizedValue[$this->secondLanguageId] = $value[self::SECOND_LANGUAGE_PLACEHOLDER];
+        }
+
+        return $localizedValue;
+    }
+}
+
+class TestableObjectModel extends ObjectModel
```

Comment:
Should be outside this file, maybe in a fixture namespace?

Reply:
Since it's only used in this test (and not meant to be used somewhere else) I like the fact that's it's in the same file

Reply:
Afaik, PSR-4 will also trigger warning because of having two classes in the same file

Reply:
I don't think so, I already did it in other tests classes But just because I like you I will move it 😘